packages/suite/src/components/suite/TorLoader/index.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect, ComponentType } from 'react';
+
+import styled from 'styled-components';
+import { TorStatus } from '@suite-types';
+import { Translation } from '@suite-components';
+import { useActions, useSelector } from '@suite-hooks';
+import { selectTorState } from '@suite-reducers/suiteReducer';
+import * as suiteActions from '@suite-actions/suiteActions';
+
+import { Button, ModalProps } from '@trezor/components';
+import { TorProgressBar } from './TorProgressBar';
+
+const StyledButton = styled(Button)`
+    width: 150px;
+`;
+
+interface TorLoadingScreenProps {
+    ModalWrapper: ComponentType<ModalProps>;
+    callback: (value: boolean) => void;
+}
+
+export const TorLoader = ({ callback, ModalWrapper }: TorLoadingScreenProps) => {
+    const [progress, setProgress] = useState<number>(0);
+    // We create a local `isDisabling` flag to make the fake disabling,
+    // since if we use Tor state, the information is real about the Tor state
+    // and we want to show user the fake loading feedback.
+    const [isDisabling, setIsDisabling] = useState<boolean>(false);
+    const { torBootstrap, isTorError } = useSelector(selectTorState);
+
+    const { toggleTor, updateTorStatus } = useActions({
+        toggleTor: suiteActions.toggleTor,
+        updateTorStatus: suiteActions.updateTorStatus,
+    });
+
+    useEffect(() => {
+        // When Tor is disabling there might still be some bootstrap event coming but
+        // we will ignore them since when disabling started there is no way back in this component
+        // We only relay on fakeProgress when disabling.
+        if (isDisabling) {
+            return;
+        }
+        if (progress === 100) {
+            setProgress(0);
+        }
+        if (torBootstrap && torBootstrap.current) {
+            setProgress(torBootstrap.current);
+            if (torBootstrap.current === torBootstrap.total) {
+                updateTorStatus(TorStatus.Enabled);
+                callback(true);
+            }
+        }
+    }, [progress, torBootstrap, callback, updateTorStatus, isDisabling]);
+
+    const tryAgain = async () => {
+        setProgress(0);
+        updateTorStatus(TorStatus.Enabling);
+
+        try {
+            await toggleTor(true);
+        } catch {
+            updateTorStatus(TorStatus.Error);
+        }
+    };
+
+    const disableTor = async () => {
+        setIsDisabling(true);
+        let fakeProgress = 0;
+        // We do not wait until toggleTor is done since we want to display fake progress.
+        toggleTor(false);
+
+        // This is a total fake progress, otherwise it would be too fast for user.
+        await new Promise(resolve => {
+            const interval = setInterval(() => {
+                if (fakeProgress === 100) {
+                    clearInterval(interval);
+                    return resolve(null);
+                }
+
+                fakeProgress += 10;
```

Comment:
cound be faster, enabling tor was quicker than disabling for me :D

Reply:
This `300` milliseconds is based in that they say that human brain requires from 200 to 300 milliseconds in order to be able to process that visual information. 
We could change it to 200, I see it should be fine with 200 as well.

Reply:
Not pushing for it to be faster it's just a bit weird that it's faster to turn it on at times than to turn it off :)